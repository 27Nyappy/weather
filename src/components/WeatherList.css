.WeatherList {
  @media only screen and (max-width: 430px) {
    padding: 20px;
    width: 80%;
    height: 250px;
  }

  @media screen and (min-width: 431px) and (max-width: 600px) {
    padding: 15px 30px;
    width: 80%;
    height: 260px;
  }

  @media screen and (min-width: 601px) and (max-width: 768px) {
    padding: 15px 30px;
    width: 80%;
    height: 270px;
  }

  @media screen and (min-width: 769px) and (max-width: 834px) {
    padding: 15px 50px 15px 30px;
    width: 75%;
    height: 280px;
  }

  @media screen and (min-width: 835px) {
    padding: 15px 50px 15px 30px;
    width: 70%;
    height: 290px;
  }

  display: flex;
  flex-direction: row;
  white-space: nowrap;
  overflow-x: scroll;
  margin: 20px auto;
  text-align: center;
  max-width: 600px;
  background-color: #1d3230c1;
  box-shadow: 0 0 3px #101414c1;
  color: #b1d4d1;

  li {
    display: inline-block;
    -webkit-overflow-scrolling: touch;
  }
}

.WeatherList::-webkit-scrollbar {
  width: 20px;
}

.WeatherList::-webkit-scrollbar-track {
      box-shadow: inset 0 0 5px grey;
      border-radius: 10px;
}

.WeatherList::-webkit-scrollbar-thumb {
  background: #b1d4d1;
  border-radius: 10px;
}

.WeatherList::-webkit-scrollbar-thumb:hover {
  background: #b1d4d1;
}