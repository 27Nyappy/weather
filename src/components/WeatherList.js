import React from 'react';
import PropTypes from 'prop-types';
import WeatherCard from './WeatherCard';

const WeatherList = ({ weatherItems }) => {
  let weatherLists = weatherItems;
  console.log(weatherItems);
  if(weatherLists.length === undefined) {
    return (
      <li>
        <WeatherCard  mainTemp={weatherLists.mainTemp} feelsLike={weatherLists.feelsLike} maxTemp={weatherLists.maxTemp} minTemp={weatherLists.minTemp} humidity={weatherLists.humidity} detail={weatherLists.detail} description={weatherLists.description} icon={weatherLists.icon} date={weatherLists.date} city={weatherLists.city} />
      </li>
    );
  }
  else {
    weatherLists = weatherItems.map((card) => {
      console.log(card);
      return (
        <li key={card.date}>
          <WeatherCard mainTemp={card.mainTemp} feelsLike={card.feelsLike} maxTemp={card.maxTemp} minTemp={card.minTemp} humidity={card.humidity} detail={card.detail} description={card.description} icon={card.icon} date={card.date} city={card.city} />
        </li>
      );
    });
  }

  return (
    <ul>
      {weatherLists}
    </ul>
  );
};

WeatherList.propTypes = {
  weatherItems: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.array
  ]).isRequired
};

export default WeatherList;