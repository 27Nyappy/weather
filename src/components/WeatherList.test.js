import React from 'react';
import { shallow } from 'enzyme';
import WeatherList from './WeatherList';

describe('Weather List Component', () => {
  const testCards = [
    {
      mainTemp: 50.13,
      feelsLike: 43.68,
      maxTemp: 53.6,
      minTemp: 46,
      humidity: 81,
      detail: 'Rain',
      description: 'moderate rain',
      icon: 'http://openweathermap.org/img/wn/10n@2x.png',
      date: 'Mon Jan 27 2020 22:57:33 GMT-0800 (Pacific Standard Time)',
      city: 'Portland'
    },
    {
      mainTemp: 50.13,
      feelsLike: 43.68,
      maxTemp: 53.6,
      minTemp: 46,
      humidity: 81,
      detail: 'Rain',
      description: 'moderate rain',
      icon: 'http://openweathermap.org/img/wn/10n@2x.png',
      date: 'Mon Jan 27 2020 24:22:35 GMT-0800 (Pacific Standard Time)',
      city: 'Portland'
    }
  ];

  it('Renders Weather List', () => {
    const wrapper = shallow(<WeatherList weatherItems={testCards} />);
    expect(wrapper).toMatchSnapshot();
  });
});