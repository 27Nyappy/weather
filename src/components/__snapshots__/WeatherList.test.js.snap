// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather List Component Renders Weather List 1`] = `
<ul
  className="WeatherList"
>
  <li
    key="Mon Jan 27 2020 22:57:33 GMT-0800 (Pacific Standard Time)"
  >
    <WeatherCard
      city="Portland"
      date="Mon Jan 27 2020 22:57:33 GMT-0800 (Pacific Standard Time)"
      description="moderate rain"
      detail="Rain"
      feelsLike={43.68}
      humidity={81}
      icon="http://openweathermap.org/img/wn/10n@2x.png"
      mainTemp={50.13}
      maxTemp={53.6}
      minTemp={46}
    />
  </li>
  <li
    key="Mon Jan 27 2020 24:22:35 GMT-0800 (Pacific Standard Time)"
  >
    <WeatherCard
      city="Portland"
      date="Mon Jan 27 2020 24:22:35 GMT-0800 (Pacific Standard Time)"
      description="moderate rain"
      detail="Rain"
      feelsLike={43.68}
      humidity={81}
      icon="http://openweathermap.org/img/wn/10n@2x.png"
      mainTemp={50.13}
      maxTemp={53.6}
      minTemp={46}
    />
  </li>
</ul>
`;
